Source/Tests/LinuxSyscalls/Syscalls.h

```diff
@@ -206,6 +218,97 @@ class SyscallHandler : public FEXCore::HLE::SyscallHandler {
   #endif
 
   std::unique_ptr<FEX::HLE::MemAllocator> Alloc32Handler{};
+  
+  ///// VMA (Virtual Memory Area) tracking /////
+
+  FEXCore::HLE::AOTIRCacheEntryLookupResult LookupAOTIRCacheEntry(uint64_t GuestAddr) final override;
+
+
+  struct MRID {
+    uint64_t dev; /* kernel only uses lower 32-bits of dev_t */
+    uint64_t id;
+
+    bool operator<(const MRID& other) const {
+      return std::tie(dev, id) < std::tie(other.dev, other.id);
+    }
+  };
+
+  // We use bits 63:32 for our own custom dev tracking
+  static constexpr const uint64_t MRID_PRIVATE = 0x1'0000'0000ULL;
+  static constexpr const uint64_t MRID_ANON = 0x2'0000'0000ULL;
+  static constexpr const uint64_t MRID_SHM =  0x3'0000'0000ULL;
+
+  struct VMAEntry;
+
+  struct MappedResource {
+    typedef std::map<MRID, MappedResource> ContainerType;
+
+    FEXCore::IR::AOTIRCacheEntry *AOTIRCacheEntry;
+    VMAEntry *FirstVMA;
+    uint64_t Length; // 0 if not fixed size
+    ContainerType::iterator Iterator;
+  };
+
+  struct VMAProt {
+    bool Writable: 1;
+    bool operator==(const VMAProt& other) const {
+      return other.Writable == Writable;
+    }
+  };
+
+  struct VMAOptions {
+    VMAOptions(int Writable, const int Shared) : Prot {Writable != 0}, Flags {Shared != 0} { }
+
+    VMAProt Prot;
+
+    const struct {
+      bool Shared: 1;
+    } Flags;
+  };
+
+  struct VMAEntry {
+    MappedResource* Resource;
+    VMAEntry* ResourcePrevVMA;
+    VMAEntry* ResourceNextVMA;
+
+    uint64_t Base;
+    uint64_t Offset;
```

Comment:
How about renaming the members to clarify this? I wouldn't have know what these variables refer to without your comment, but this would help:
```cpp
uint64_t VmaBase;
uint64_t ResourceOffset; // relative to VmaBase
uint64_t VmaLength;
```

Reply:
The ResourceOffset is actually not relative to the VmaBase, it's just the offset to the mapped resource.

I don't really think renaming these makes (the already) complex logic any simpler, but it does make the code substantially harder to read with longer variable names, with `vma->VmaBase` all over and such.

I'd be fine with 
```
uint64_t Offset; // to the associated MRID, if any
```

I think, as with many such cases which are very esoteric and bound to change often, it's best to have an overview of the implementation, and if unsure ask the people involved in that module for a quick intro / help, rather than to overburden the code by trying to make it easier to understand, which is unlikely to succeed anyway.

Reply:
> The ResourceOffset is actually not relative to the VmaBase, it's just the offset to the mapped resource.

... so again, what is it the offset *from*? What's the base that this offset is added to, if it's not the `Base` member right above?

Reply:
It's the offset to the mapped resource, ie, *vma->base would return the contents of MappedResource::read(Offset), the same way to mmap addresses relate to the offset of fds.

Reply:
Your use of the phrase `offset to the mapped resource` throws me off. Grammatically speaking, an offset *to* X would be an offset that points to *the start of* X, but it sounds like you're talking to an offset *in* X (i.e. something that skips the first `Offset` bytes of X)?

In particular for this case, do I understand it correctly that `Length` bytes of the `MappedResource` (starting from `Offset`) are mapped at `vma->base`?

(If this is not the case, can you elaborate your mmap comparison a bit, explicitly highlighting what the FD and the addresses compare to?)

Reply:
In my head it is the 'offset to the corresponding backing page from the beginning of the backing storage', so whatever. It is extremely common even for native english speakers to misuse of/in/to and I think the meaning is pretty obvious from the context, name, intended functionality, etc either way.

Reply:
> I think the meaning is pretty obvious

It really is not though, otherwise I wouldn't have asked. Until my fourth comment I didn't understand the relation of the three members, and even then I was still unsure. Anyway, proceed with this information as you please.